beacon-chain/blockchain/process_block.go

```diff
@@ -529,6 +540,22 @@ func (s *Service) lateBlockTasks(ctx context.Context) {
 		log.WithError(err).Debug("could not update next slot state cache")
 	}
 
+	// Update caches when across epoch boundary
+	if s.CurrentSlot()%params.BeaconConfig().SlotsPerEpoch == params.BeaconConfig().SlotsPerEpoch-1 {
+		e := slots.ToEpoch(headState.Slot())
+		epochStartSlot := e.Mul(uint64(params.BeaconConfig().SlotsPerEpoch))
+		boundarySlot := primitives.Slot(epochStartSlot) + params.BeaconConfig().SlotsPerEpoch - 1
+		st := headState.Copy()
+		st, err := transition.ProcessSlotsIfPossible(ctx, st, boundarySlot)
```

Comment:
again, this processes slots twice ? Since you process it here and again in `updateEpochBoundaryCaches`